src/main/java/com/ibm/eventstreams/connect/mqsource/builders/BaseRecordBuilder.java

```diff
@@ -74,18 +78,27 @@ else if (kh.equals(MQSourceConnector.CONFIG_VALUE_MQ_RECORD_BUILDER_KEY_HEADER_J
             }
         }
 
-        log.trace("[{}]  Exit {}.configure", Thread.currentThread().getId(), this.getClass().getName());
-    }
+		String str = props.get(MQSourceConnector.CONFIG_NAME_MQ_JMS_PROPERTY_COPY_TO_KAFKA_HEADER);
+
+        if (Optional.of(str).isPresent()) {
```

Comment:
Just replace with a simple
`if (str != null) {`

The way you've done it, I get a NullPointerException if I omit the new config option from my configuration,

Reply:
i apologize, i intended to use ofNullable#orElse method chain to avoid both NPE & if/null checks. fixed and tested fine.